src/lib.rs

```diff
@@ -272,6 +272,31 @@ pub struct PaintCtx<'a, 'b: 'a> {
     /// The render context for actually painting.
     pub render_ctx: &'a mut Piet<'b>,
     pub window_id: WindowId,
+    /// The currently visible region.
+    pub(crate) region: Region,
+}
+
+/// A region of a widget, generally used to describe what needs to be drawn.
+#[derive(Debug, Clone)]
+pub struct Region(Rect);
+
+impl Region {
+    /// Returns the smallest `Rect` that encloses the entire region.
+    pub fn to_rect(&self) -> Rect {
+        self.0
+    }
+
+    /// Returns `true` if `self` intersects with `other`.
+    #[inline]
+    pub fn intersects(&self, other: Rect) -> bool {
+        self.0.intersect(other).area() > 0.
+    }
+}
```

Comment:
Probably not essential for this PR, but I think implementing Add/Sub<Vec2> would be useful.

Also I think it would be slightly better if `intersect` were on here too, rather than forcing going through `Rect`.

Reply:
What does `intersect` return? Another region? My inclination was to keep this as minimal as possible for now, and then when we want to implement minimal invalidation, we can flesh out exactly what this should look like.

Reply:
I'm thinking another region, yes. And yeah, we don't need to get too fancy when region is just rect. But I can definitely imagine having more fine-grained regions in the future.